### PR TITLE
Welling/survey upload support

### PR DESCRIPTION
**Pull request by jswelling**
Adds support for Uploads to survey.py and the utilities like split_and_create that use it.

### CODE REVIEW

File: src/ingest-pipeline/misc/tools/survey.py [hunk 1]
```diff
@@ -198,15 +211,46 @@ def __init__(self, prop_dct, entity_factory):
...
         if isinstance(c_h_g, str):
             c_h_g = (c_h_g.lower() in ['yes', 'true'])
         self.contains_human_genetic_sequences = c_h_g
         self._kid_dct = None
         self._parent_dct = None
-    
+        self._organs = None
+
+    @property
+    def donor_uuid(self):
```
Comment (jpuerto-psc) [inline src/ingest-pipeline/misc/tools/survey.py:225]: I don't quite understand the difference in this if/else case (probably b/c I don't know how this data is structured). Is this meant to be able to handle the case where a parent entity is not marked as a Donor, but does have a donor_uuid associated with it?

File: src/ingest-pipeline/misc/tools/survey.py [hunk 2]
```diff
@@ -325,15 +376,164 @@ def all_dataset_uuids(self):
...
+        if self._organ is None:
+            # There should only be one parent, another Sample
+            for parent_uuid in self.parent_uuids:
+                parent_entity = self.entity_factory.get(parent_uuid)
+                assert isinstance(parent_entity, Sample), 'Outermost sample has no organ code?'
+                self._organ = parent_entity.organ
+        return self._organ
+
+    @property
+    def donor_uuid(self):
```
Comment (jpuerto-psc) [inline src/ingest-pipeline/misc/tools/survey.py:508]: Won't this always break on the first parent_uuid in the list?

File: src/ingest-pipeline/misc/tools/survey.py (hunk 1 above)
Comment (jswelling) [reply src/ingest-pipeline/misc/tools/survey.py:225]: Ah.  That's awkwardly coded, but the point is,
* if the parent is a Donor, it's uuid is the thing we want.
* if not, the parent is a Dataset or a Sample, in which case evaluating parent_entity.donor_uuid (with no leading underscore) will recursively call its donor_uuid property, causing it to check *its* parent.  So it's recursion with a property call each time, until it reaches the Donor and the first clause executes.  But we fill in all the _donor_uuid properties on the way down, so they are available in the cache.

File: src/ingest-pipeline/misc/tools/survey.py (hunk 2 above)
Comment (jswelling) [reply src/ingest-pipeline/misc/tools/survey.py:508]: Yeah, I agree :-/ .  I'll restructure that.

**Pull request merged**